Sources/SwifterAuth.swift

```diff
@@ -76,28 +53,39 @@ public extension Swifter {
                    failure: FailureHandler? = nil) {
         let callbackURLScheme = callbackURL.absoluteString.components(separatedBy: "://").first
         self.postOAuthRequestToken(with: callbackURL, success: { token, response in
-            var requestToken = token!
             let queryURL = self.makeQueryURL(tokenKey: token!.key, forceLogin: forceLogin)
-            let session = ASWebAuthenticationSession(url: queryURL, callbackURLScheme: callbackURLScheme) { (url, error) in
+            self.session = ASWebAuthenticationSession(url: queryURL, callbackURLScheme: callbackURLScheme) { (url, error) in
                 if let error = error {
                     failure?(error)
                     return
                 }
-                let parameters = url!.query!.queryStringParameters
-                guard let verifier = parameters["oauth_verifier"] else {
-                    let error = SwifterError(message: "User cancelled login from Twitter App", kind: .cancelled)
-                    failure?(error)
-                    return
-                }
-                requestToken.verifier = verifier
-                self.postOAuthAccessToken(with: requestToken, success: { accessToken, response in
-                    self.client.credential = Credential(accessToken: accessToken!)
-                    success?(accessToken!, response)
-                }, failure: failure)
+                self.postOAuthAccessTokenHelper(requestToken: token!, responseURL: url!, success: success, failure: failure)
+            }
+            self.session?.presentationContextProvider = provider
+            self.session?.prefersEphemeralWebBrowserSession = ephemeralSession
+            self.session?.start()
+        }, failure: failure)
+    }
+    #endif
+
+    /**
+     Begin Authorization with a Callback URL.
+     - OS X only
+     */
+    #if os(macOS)
+    @available(macOS, deprecated: 10.15)
```

Comment:
Does this really have to be deprecated?

Reply:
This wasn't Apple's spec, it was my personal intention.
I corrected `available` annotations by commit 5ca503c.